dGame/dComponents/PropertyEntranceComponent.h

```diff
@@ -59,15 +61,17 @@ class PropertyEntranceComponent : public Component {
 	 */
 	[[nodiscard]] LWOMAPID GetMapID() const { return m_MapID; };
 
-	PropertySelectQueryProperty SetPropertyValues(PropertySelectQueryProperty property, LWOCLONEID cloneId = LWOCLONEID_INVALID, std::string ownerName = "", std::string propertyName = "", std::string propertyDescription = "", float reputation = 0, bool isBFF = false, bool isFriend = false, bool isModeratorApproved = false, bool isAlt = false, bool isOwned = false, uint32_t privacyOption = 0, uint32_t timeLastUpdated = 0, float performanceCost = 0.0f);
-
-	std::string BuildQuery(Entity* entity, int32_t sortMethod, Character* character, std::string customQuery = "", bool wantLimits = true);
+	void PopulateUserFriendMap(uint32_t user);
 
+	std::vector<uint32_t> GetPropertyIDsBasedOnParams(const std::string& searchText, uint32_t sortMethod, Entity* requestor);
+	
+	PropertyData GetPropertyData(uint32_t propertyID);
+	PropertyPersonalData GetPropertyPersonalData(PropertyData& propertyData, Entity* queryingUser, bool updatePropertyDataStructure = false);
 private:
-	/**
-	 * Cache of property information that was queried for property launched, indexed by property ID
-	 */
-	std::map<LWOOBJID, std::vector<PropertySelectQueryProperty>> propertyQueries;
+	std::unordered_map<LWOOBJID, std::vector<uint32_t>> m_UserRequestedCloneMap;
+	std::unordered_map<uint32_t, PropertyData> m_PropertyDataCache;
+
+	std::unordered_map<LWOOBJID, std::unordered_map<LWOOBJID, bool>> m_UserFriendMap;
```

Comment:
an unordered map of bool can be exchanged with unordered set to save memory and readability, however there are concerns above on if this can even be used due to this data not being updated after the initial query.  The same applies for the propertyDataCache. Are both of these updated if there is a change in the users friend list or if there is a change in the property status?  for example if you removed a friend they could visit your property still if you never update the cache here until the world shutsdown, and someone could make an unapproved property and then fly to it because the property was approved previously.  Are these concerns resolved somewhere else that I missed?